File: aesi-eclipse/bundle/src/main/java/com/virtlink/editorservices/eclipse/AesiModule.java
package com.virtlink.editorservices.eclipse;

import org.eclipse.jface.text.contentassist.IContentAssistProcessor;

import com.google.inject.AbstractModule;
import com.google.inject.Singleton;
import com.google.inject.assistedinject.FactoryModuleBuilder;
import com.virtlink.aesi.eclipse.DocumentManager;
import com.virtlink.aesi.eclipse.ProjectManager;
import com.virtlink.aesi.eclipse.codecompletion.AesiCompletionProcessor;
import com.virtlink.aesi.eclipse.editors.AesiSourceViewerConfiguration;
import com.virtlink.aesi.eclipse.editors.ColorManager;
import com.virtlink.aesi.eclipse.editors.IContentAssistProcessorFactory;

public class AesiModule extends AbstractModule {

	@Override
	protected void configure() {
		bind(ProjectManager.class).in(Singleton.class);
		bind(DocumentManager.class).in(Singleton.class);
		bindCodeCompletion();
	}
	
	protected void bindCodeCompletion() {
		install(new FactoryModuleBuilder()
				.implement(IContentAssistProcessor.class, AesiCompletionProcessor.class)
				.build(IContentAssistProcessorFactory.class));
		bind(ColorManager.class).toInstance(new ColorManager());
		bind(AesiSourceViewerConfiguration.class);
	}

}

File: aesi-intellij/src/main/kotlin/com/virtlink/editorservices/intellij/AesiModule.kt
package com.virtlink.editorservices.intellij

import com.google.inject.AbstractModule
import com.google.inject.Singleton
import com.intellij.psi.tree.IFileElementType
import com.google.inject.assistedinject.FactoryModuleBuilder
import com.intellij.lexer.Lexer
import com.virtlink.editorservices.content.IContentManager
import com.virtlink.editorservices.intellij.content.DocumentContentManager
import com.virtlink.editorservices.intellij.content.IntellijContentSource
import com.virtlink.editorservices.intellij.content.TextContentSource
import com.virtlink.editorservices.intellij.psi.*
import com.virtlink.editorservices.intellij.referenceresoluton.AesiReferenceProvider
import com.virtlink.editorservices.intellij.syntaxcoloring.AesiLexer
import com.virtlink.editorservices.intellij.syntaxcoloring.ILexerFactory
import com.virtlink.editorservices.intellij.syntaxcoloring.ScopeManager


abstract class AesiModule : AbstractModule() {
    override fun configure() {
        bind(DocumentManager::class.java).`in`(Singleton::class.java)
        bind(ProjectManager::class.java).`in`(Singleton::class.java)
        bind(AesiTokenTypeManager::class.java).`in`(Singleton::class.java)
        bind(AesiElementTypeManager::class.java).`in`(Singleton::class.java)
        bind(ScopeManager::class.java).`in`(Singleton::class.java)
        bind(IFileElementType::class.java).to(AesiFileElementType::class.java).`in`(Singleton::class.java)
        bind(AesiReferenceProvider::class.java).`in`(Singleton::class.java)
        bind(IContentManager::class.java).to(DocumentContentManager::class.java).`in`(Singleton::class.java)
        bind(IntellijContentSource::class.java).`in`(Singleton::class.java)
        bind(TextContentSource::class.java).`in`(Singleton::class.java)
        install(FactoryModuleBuilder()
                .implement(Lexer::class.java, AesiLexer::class.java)
                .build(ILexerFactory::class.java))
        install(FactoryModuleBuilder()
                .implement(AesiAstBuilder::class.java, AesiAstBuilder::class.java)
                .build(IAstBuilderFactory::class.java))
        bindCodeCompletion()
        bindSyntaxColoring()
    }

    protected open fun bindCodeCompletion() {
    }

    protected open fun bindSyntaxColoring() {

    }
}